pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -130,17 +128,46 @@ func (d *DependenciesDistributor) NeedLeaderElection() bool {
 
 func (d *DependenciesDistributor) discoverResources(period time.Duration) {
 	wait.Until(func() {
-		for _, gvr := range supportedTypes {
-			if d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
-				continue
+		newResources := lifted.GetDeletableResources(d.DiscoveryClientSet)
```

Comment:
We should only support namespaced resources.

Reply:
Currently, only namespace scope resources are supported. I am working on implementing cluster scope resources.

Reply:
I think the community has discussed this before. More Info: https://github.com/karmada-io/karmada/pull/3044.
In addition, we should limit the type of resources in namespaced resources. Only resources that can be resolved into dependencies should be enqueued, instead of all resources, which will affect performance.

Reply:
Thank you for the reminder.